### PR TITLE
Mini aod 2018

### PR DESCRIPTION
**Pull request by fanfanman**
The ADD feature is merged into runAnalysis.py to avoid redundant files.
The commands to process ADD files are:
> python runAnalysis.py --add2016 -s
> python runAnalysis.py --add2016 -s -e

### CODE REVIEW

File: test/runAnalysis/runAnalysis.py [hunk 1]
```diff
@@ -245,6 +257,9 @@ def main():
 		arguments["year"] = 2018
 	cmssw_cfg = open('setup.py').read()%arguments
 	prefix = "muons_"	
+	if args.add2016:
+		cmssw_cfg = cmssw_cfg + "\nfrom Configuration.AlCa.GlobalTag import GlobalTag \nprocess.GlobalTag = GlobalTag(process.GlobalTag, 'auto:startup', '') \n"
```
Comment (JanFSchulte) [inline test/runAnalysis/runAnalysis.py:261]: Again the question, why do you do this?

File: test/runAnalysis/runAnalysis.py [hunk 2]
```diff
@@ -310,8 +325,12 @@ def main():
 					from samples import backgrounds_muons_2018 as samples
 				else:	
 					from samples import backgrounds_muons_2017 as samples 
+			if args.add2016:
+				from samples import add_2016 as samples
 			lumi_mask = ""
 			GT = "94X_mc2017_realistic_v14"
+			if args.add2016:
+				GT = "84X_mcRun2_asymptotic_2016_TranchelV_v6"
```
Comment (JanFSchulte) [inline test/runAnalysis/runAnalysis.py:333]: As far as I am aware our processing of the miniAOD files does nothing were the information contained in the global tags is actually used, so this doesn't really matter. But we should in principle keep in mind that we should add here the correct GT configuration for all the years of data taking. Also, I'm pretty sure there is no CMSSW 84X, so I have no idea what kind of tag you picked here ;)
But in general, could you have a look at this table and add all the GTs for data and MC analysis here https://twiki.cern.ch/twiki/bin/viewauth/CMS/PdmVAnalysisSummaryTable
If there are two, please pick the second one.

File: test/runAnalysis/runAnalysis.py [hunk 3]
```diff
@@ -358,10 +377,27 @@ def main():
...
 				open('cmssw_cfg.py', 'wt').write(cmssw_tmp)
             			if args.submit:
-                			os.system('crab submit -c crabConfig.py')
+                			#os.system('crab submit -c crabConfig.py')
+					os.system('cmsRun cmssw_cfg.py')
+					fnamere = "dileptonAna_%s%s.root"%(prefix, dataset_name)
+                                        if "muon" in fnamere: fnamere = fnamere.replace("LL", "2Mu")
+                                        else: fnamere = fnamere.replace("LL", "2E")
+                                        os.system('mv zp2mu_histos.root %s'%fnamere)
+                                        #print "Test passed for: " + dataset_name
```
Comment (JanFSchulte) [inline test/runAnalysis/runAnalysis.py:400]: I'm pretty sure all this stuff here breaks the submission to CRAB! I think it is a neat implementation on how to run on a sample locally, but that should be an extra feature and not break existing functionality!

File: test/runAnalysis/runAnalysis.py [hunk 4]
```diff
@@ -56,6 +56,15 @@ def getFilterSnippet(name,args,doApply=False,year=2017):
...
 	getattr(process,path_name).insert(2,process.DYGenMassFilter)'''
+        ADDFilter1700 = '''
+process.DYGenMassFilter = cms.EDFilter('QScaleSelector',
+                                       src = cms.InputTag('generator'),
+                                       min_mass = cms.double(1700),
+                                       max_mass = cms.double(7000)
+                                       )
+for path_name, path in process.paths.iteritems():
+        getattr(process,path_name).insert(2,process.DYGenMassFilter)'''
+
```
Comment (JanFSchulte) [inline test/runAnalysis/runAnalysis.py:67]: Why is this necessary? The lower cut of 1700 is baked into the sample and you don't have to apply it here. And why do you an upper cut of 7000?

File: test/runAnalysis/runAnalysis.py (hunk 2 above)
Comment (fanfanman) [reply test/runAnalysis/runAnalysis.py:333]: Thanks a lot! I have just fixed this!

File: test/runAnalysis/runAnalysis.py (hunk 4 above)
Comment (fanfanman) [reply test/runAnalysis/runAnalysis.py:67]: You are right! I just checked the files with and without this cut, they are actually the same!
